### DISCUSSION TITLE
Improvements and concerns before merging daisyUI branch

### DISCUSSION THREAD

**Discussion by saadeghi** [yesvelte/yesvelte#297] [General]
@pournasserian Here are the things we need to discuss moving forward.

1. There are many duplicate items in sidebar here: https://www.yesvelte.com/docs I think we should remove duplicate items
2. Documents site has a trailing dash problem. for example this https://www.yesvelte.com/docs works. but this https://www.yesvelte.com/docs/ doesn't work. I think it can be fixed using `export const trailingSlash = "always"` in `+layout.js` 
3. The way `steps` style is being handled is not efficient and there would be an item limit for the style. I thin this style iscoming from Tabler but currently, the way it works is it applies active color to all steps and the applies inactive color to the item after the active item and the next one and the next one... and it's literally a huge series of CSS `~` selectors:  
    <img width="1462" alt="Screenshot 2023-04-04 at 07 56 09" src="https://user-images.githubusercontent.com/7342023/229691079-ed59fa7e-fb87-4683-986a-539a724f3ce6.png">
    It also limits the flexibility of step items. For example we won't be able to set a different color to a step (let's say step 1 and 2 is passed and there's an error in step 3. it should be error color). Currently it only works as a progressbar and shows the current step. I think daisyUI's approach is more flexible and also generates more efficient style. we can add colors to each step instead coloring all previous steps.
4. `slider` component is using `noUi-` prefix instead of `y-`. Is this going to be fixed? or should I use the same selectors and if it's coming from a plugin, should I import that style directly?
5. `autocomplete` component is using `ts-` prefix and also has some classnames without prefix (like `single`) Is this going to be fixed? or should I use the same selectors and if it's coming from a plugin, should I import that style directly?
6. Same question about Datepicker.
7. In [this example](https://www.yesvelte.com/docs/form/form-fieldset#Layout) it generates and empty `y-switch-label` tag. I think if that `<label>` tag is not going to have text in it, it shouldn't be rendered.
8. There are size differences for some components. For example daisyUI's default `toggle` (switch) size is larger than Tabler. Should I change the default size? or should I keep it the way it is?
9. I noticed some new components are now added in the main branch. I will add them as well after the merge.
10. There is currently a bug in daisyUI's `input-group` which applies wrong border radius if there's only one item in the group. I haven't notice this before because it was always more than 1 item in the group. However there are many places in Yesvelte examples with 1 input inside a wrapper. I will fix this soon on my side and then I update the dependency.

**Comment by TheHadiAhmadi** [comment]
3- for steps your suggestion is better than current implementation, but what do you think about something like [this](https://www.yesvelte.com/docs/progress#Group):

```svelte
<Steps>
    <StepItem />
    <StepItem />
    <StepItem color="blue"/>
    <StepItem color="red"/>
    <StepItem />
    <StepItem />
</Steps>
``` 
1 -> blue
2 -> blue
3 -> blue
4 -> red
5 -> default (gray)
6 -> default (gray)

7- probably there is a bug in Switch or FormSwitch component which renders empty `<label>`, I'll check it.
